python/lsst/jointcal/dataIds.py

```diff
@@ -102,7 +102,7 @@ def makeDataRefList(self, namespace):
                     self._addDataRef(namespace, ref.dataId, tract)
 
         # Ensure all components of a visit are kept together by putting them all in the same set of tracts
-        for visit, tractSet in visitTract.items():
+        for visit, tractSet in sorted(visitTract.items()):
```

Comment:
I checked this for py2 vs py3, and if you want the dataRefs to be consistently ordered I think you need to also have sorted(tractSet) below.  Otherwise I think this is only sorting visits.